app/controllers/api/physical_servers_controller.rb

```diff
@@ -52,26 +52,47 @@ def refresh_resource(type, id, _data = nil)
       end
     end
 
-    private
-
-    def change_resource_state(state, type, id)
-      raise BadRequestError, "Must specify an id for changing a #{type} resource" unless id
+    def apply_config_pattern_resource(type, id, data)
+      change_resource_state(:apply_config_pattern, type, id, data)
+    end
 
-      ensure_resource_exists(type, id) if single_resource?
+    private
 
+    def change_resource_state(state, type, id, data = {})
+      validate_resource(state, type, id, data)
       api_action(type, id) do |klass|
         begin
           server = resource_search(id, type, klass)
+          ensure_customization_script_exists(data["pattern_id"]) if apply_config_action(state)
           desc = "Requested server state #{state} for #{server_ident(server)}"
           api_log_info(desc)
           task_id = queue_object_action(server, desc, :method_name => state, :role => :ems_operations)
           action_result(true, desc, :task_id => task_id)
-        rescue => err
+        rescue StandardError => err
           action_result(false, err.to_s)
         end
       end
     end
 
+    def validate_resource(state, type, id, data = {})
+      raise BadRequestError, "Must specify an id for changing a #{type} resource" unless id
+      if single_resource?
+        ensure_resource_exists(type, id)
+        # Verify the configuration pattern resource if the operation is apply_config_pattern
+        ensure_customization_script_exists(data["pattern_id"]) if apply_config_action(state)
```

Comment:
I see that this is being called twice now within `change_resource_state`, both here and in the method above (`change_resource_state`). WDYT about moving this check to the `apply_config_pattern_resource` method above, since it seems to be specific to this method? That will remove the need to check `if apply_config_action`

Reply:
But, the problem is that moving this check to apply_config_pattern_resource method It won't works if not single resource. For the operation in multiple resources is necessary to have a check inside the api_action iteration.